pkg/postage/batchservice/batchservice.go

```diff
@@ -261,7 +264,14 @@ func (svc *batchService) Start(startBlock uint64, initState *postage.ChainSnapsh
 		startBlock = initState.LastBlockNumber
 	}
 
-	return svc.listener.Listen(startBlock+1, svc, initState), nil
+	syncedChan := svc.listener.Listen(startBlock+1, svc, initState)
+
+	select {
+	case err = <-syncedChan:
+		return err
+	case <-interrupt:
```

Comment:
something to bear in mind here is that once the `os.Signal` is read off the channel, you must make sure that the error bubbles up correctly. it might be a bit risky to pass this channel down the stack, since there are multiple points where we potentially read from it. the signal is sent once, rather than a closure of the channel, so if you read from it here, some other component might not be aware of the signal. so for example, since this in a light node will be done in a separate goroutine, the syncing process fails, but the process has been sent a kill signal, however (maybe) this goroutine picks up the signal, but the error does not bubble up the stack, and the process never exists (except for when you signal it with the kill or term _again_, leading the value to be read again from the same channel by a different goroutine). i would suggest not to pass this channel between components, but rather wrap it in `cmd/start.go` in a separate goroutine with a closure that would just `close()` a fresh `chan struct{}` which you could pass down into the new bee constructor.

Reply:
Implemented a `struct{}` based channel listening on the system calls, as you described.